sklearn/multi_label/classifier_chain.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from ..base import BaseEstimator, clone
+from sklearn.linear_model import LogisticRegression
+import random
+
+
+class ClassifierChain(BaseEstimator):
+    """Classifier Chain
+    A multi-label model that arranges classifiers into a chain.
+    Predictions are made in the order specified by the chain.
+    The predictions of earlier models are used as features by later models.
+
+    By default the order of the chain is random although it can be specified
+    at the time of fitting with the chain_order parameter. Since the optimal
+    chain_order is not known a priori it is common to use the mean prediction
+    of an ensemble randomly ordered classifier chains.
+
+    Parameters
+    ----------
+    base_estimator : object
+        The base estimator used for fitting the model for each label.
+        Defaults to LogisticRegression
+    random_state : int
+        An integer used to seed the random number generator
+
+    Attributes
+    ----------
+    classifiers_ : array
+        List of classifiers, which will be used to chain prediction.
+    chain_order : list of ints
+        A list of integers specifying the order of the classes in the chain.
+        For example, for a chain of length 5
+            chain_order = [1, 3, 2, 4, 0]
+        means that the first model in the chain will make predictions for
+        column 1 in the Y matrix, the second model will make predictions for
+        column 3, etc.
+        If chain_order is not None it must have a length equal to the
+        number of columns in Y.
+        If chain_order is None an ordered list of integers will be used
+            chain_order = [0, 1, 2, ..., Y.shape[1]]
+        where Y is the label matrix passed to the fit method.
+
+    References
+    ----------
+    Jesse Read, Bernhard Pfahringer, Geoff Holmes, Eibe Frank,
+        "Classifier Chains for Multi-label Classification", 2009.
+
+    """
+
+    def __init__(self,
+                 base_estimator=LogisticRegression(),
+                 random_state=None):
+
+        self.base_estimator = base_estimator
+        random.seed(random_state)
+        self.classifiers = []
+        self.chain_order = None
+
+    def fit(self, X, Y, chain_order=None, shuffle=True):
```

Comment:
For what it's worth the convention of capitalizing the variable names of multi-label arrays already appears in the documentation here:
http://scikit-learn.org/stable/auto_examples/plot_multilabel.html#sphx-glr-auto-examples-plot-multilabel-py